packages/flutter/test/material/expansion_tile_test.dart

```diff
@@ -109,15 +109,15 @@ void main() {
     expect(getHeight(topKey), getHeight(collapsedKey) - 2.0);
     expect(getHeight(topKey), getHeight(defaultKey) - 2.0);
 
-    BoxDecoration expandedContainerDecoration = getContainer(expandedKey).decoration! as BoxDecoration;
+    ShapeDecoration expandedContainerDecoration = getContainer(expandedKey).decoration! as ShapeDecoration;
     expect(expandedContainerDecoration.color, Colors.red);
-    expect(expandedContainerDecoration.border!.top.color, dividerColor);
-    expect(expandedContainerDecoration.border!.bottom.color, dividerColor);
+    expect((expandedContainerDecoration.shape as Border?)!.top.color, dividerColor);
+    expect((expandedContainerDecoration.shape as Border?)!.bottom.color, dividerColor);
```

Comment:
if it is a test, you can just make `...as Border).top`. Don't need to cast as nullable, then !